odbc/src/api/connect_attr_tests.rs

```diff
@@ -0,0 +1,281 @@
+mod unit {
+    use crate::{
+        errors::ODBCError,
+        handles::definitions::{Connection, ConnectionState, MongoHandle},
+        util::input_wtext_to_string,
+        SQLGetConnectAttrW, SQLSetConnectAttrW,
+    };
+    use odbc_sys::{ConnectionAttribute, Integer, Pointer, SqlReturn, UInteger};
+    use std::sync::RwLock;
+
+    // Test getting CurrentCatalog attribute from the Connection
+    // when none of the attributes are set.
+    #[test]
+    fn get_string_attrs_default() {
```

Comment:
[note] I am not a big fan of the code duplication here. I think Oliver's work on the helpers could help with refactoring it later.

Reply:
To be clear, are you suggesting using the `verify_sql_diagnostics` helper method?

Reply:
I investigated how to use that function but it required quite a bit of overhead to even make it available in the tests, and even then it was as cleanly applicable as I would have expected. I did do a bunch of major refactoring on the tests to cut down on the duplication. It abstracts away a lot of the nitty-gritty details from the tests themselves, which _should_ be more useful? lmk what you think!

Reply:
uhm, yes, it looks like Oliver's helper are meant more for the integration tests than the unit tests. 
I was more thinking of a generic function for initializing and handling the recurrent code for both get and set which is exactly what you did 👍